### PR TITLE
Fixes #24093 - Notification is sent only when disk > 90%

### PR DESCRIPTION
**Pull request by dLobatog**
Here’s what happens for a proxy where the disk 12% full:

When the code runs, percentage 12% is < 90%.
The notification does not exist, therefore:

```ruby
percentage[0..2].to_i < 90 && notification_already_exists?(proxy)
true && false
false # so it does not skip the loop
```

The next line tries to update old notifications. Since
there are no notifications at this point,
update_notifications(proxy).empty? is true.

Finally it creates the notification (at this point we
have a notification for proxy ‘foreman.example.com’ full at 12%).

After that, the job is scheduled, and when it runs (12h later):

The percentage is less than 90, 12%, and the notification
already exists. At this point Foreman is happy and proceeds
to remove the notification.

```ruby
 percentage[0..2].to_i < 90 && notification_already_exists?(proxy)
 # evaluates to
 true && true
```

For the next 12h, we don’t have the notification.
Now, 12h later… GOTO beginning of the post :wink:
Basically right before creating the notification I added a check
for > 90% full. The rest of the logic should be fine.

### CODE REVIEW

**Comment by stbenjam** [comment]
[test katello]

**Comment by stbenjam** [comment]
Looks like your change fixes it, but do you think the 3 `next`'s make it harder to understand this code? It does at least to me :-\

If I understand how this works right, an if/elsif might be better...

```ruby
def deliver!
  SmartProxy.unscoped.with_content.each do |proxy|
    percentage = proxy.statuses[:pulp].storage['pulp_dir']['percent'][0..2].to_i

    if percentage < 90 && notification_already_exists?(proxy)
      blueprint.notifications.where(subject: proxy).destroy_all
    elsif update_notifications(proxy).empty? && percentage > 90
        ::Notification.create!(
          :subject => proxy,
          :initiator => User.anonymous_admin,
          :audience => Notification::AUDIENCE_ADMIN,
          :message => ::UINotifications::StringParser.new(
            blueprint.message,
            :subject => proxy,
            :percentage => "#{percentage}%"
          ),
          :notification_blueprint => blueprint
        )
   end
  end
end
```

I believe there's also another bug in `notification_already_exists?`, where it assumes that there exists at most 1 notification with the proxy as a subject, and it could be simplified to just:

```ruby
def notification_already_exists?(subject)
  blueprint.notifications.where(subject: subject).any?
end
```

**Comment by johnpmitsch** [comment]
+1 to not using all of the `next`s, which makes this hard to grok and leads to duplicate logic

**Comment by stbenjam** [comment]
Thanks @dLobatog !
